packages/flutter/lib/src/services/keyboard_key.dart

```diff
@@ -171,8 +171,8 @@ class LogicalKeyboardKey extends KeyboardKey {
     if (other.runtimeType != runtimeType) {
       return false;
     }
-    final LogicalKeyboardKey typedOther = other;
-    return keyId == typedOther.keyId;
+    return other is LogicalKeyboardKey
```

Comment:
After the previous if we know that other is assignable to LogicalKeyboardKey, so there shouldn't be a need to check that again? I think this would be clearer as:
```dart
final LogicalKeyboardKey typedOther = other as LogicalKeyboardKey;
return keyId == typedOther.keyId;
```